### PR TITLE
Changes to send mail via smtp port 465

### PR DESCRIPTION
**Pull request by IamPrvn**
AlertManager currently is not able to send mails to smtp port 465. Adding this change that will take care of the above. In alertmanager conf file, you need to explicitly supply "require_tls" to false, in order to work with smtp port 465 

Refer issue #705 

@stuartnelson3  @beorn7

### CODE REVIEW

**Comment by IamPrvn** [comment]
@stuartnelson3 @beorn7 is there any update on this?

**Comment by beorn7** [comment]
@stuartnelson3 Could you have a look? I have not much state on this.

**Comment by stuartnelson3** [comment]
Thanks for the pr! Could you run your code through `goimports` and `gofmt`? Seems like some formatting is off.

**Comment by IamPrvn** [comment]
Fixed formatting issues in the previous commit.

File: notify/impl.go [hunk 1]
```diff
@@ -254,19 +254,32 @@ func (n *Email) auth(mechs string) (smtp.Auth, error) {
...
+	var c *smtp.Client
+	host, port, err := net.SplitHostPort(n.conf.Smarthost)
 	if err != nil {
 		return false, fmt.Errorf("invalid address: %s", err)
 	}
 
+	if port == "465" {
+		conn, err := tls.Dial("tcp", n.conf.Smarthost, &tls.Config{ServerName: host})
+		if err != nil {
+			return true, err
```
Comment (stuartnelson3) [inline notify/impl.go]: `true` should be `false` in all these error blocks

Comment (IamPrvn) [inline notify/impl.go]: AFAIK, true/false flag is tied to retry logic (whether the error is recoverable or not). For the communications happening over wire, multiple attempts shall be made(i.e. error shall be deemed to be recoverable, in this case we are trying to establish a connection to remote smtp server). Infact, even before my change most of the error blocks returned true.

Comment (stuartnelson3) [inline notify/impl.go]: Ah sorry, I should have looked further. Disregard.

Comment (IamPrvn) [inline notify/impl.go]: @fabxc @juliusv @beorn7 any updates?

**Comment by beorn7** [comment]
I haven't worked with SMTP servers in Go either. @stuartnelson3 I'd say if you like it and nobody objects, go for it.

**Pull request merged**